### ISSUE TITLE
CA1868: don't trigger for ifs with an else clause

### ISSUE THREAD

**Issue by Eisfunke** [dotnet/roslyn-analyzers#7683]
### Analyzer

**Diagnostic ID**: [CA1868](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1868)

### Describe the improvement

CA1868 triggers if adds on sets or removes on collections are guarded by an if(contains) check, which isn't necessary. However, this isn't the case if there's an `else` clause, for example if you want to implement a "toggle":

```
List<string> example = ["x", "y"];

if (example.Contains("x"))  // triggers CA1868
{
    example.Remove("x");
}
else
{
    example.Add("x");
}
```

In this case, the rule shouldn't trigger as the `if(contains)` isn't *only* there to guard the remove call, but contains additional logic.

### Describe suggestions on how to achieve the rule

Don't trigger CA1868 for ifs with an else/else if clause.

**Comment by mpidash** [comment]
I don't think this is a false positive because you can still eliminate one lookup by using the return value of `Remove`, for example:

```c#
if (!example.Remove("x"))
{
    example.Add("x");
}
```
